### PR TITLE
e2e: fix potential race between controller and assertion check

### PR DESCRIPTION
**Pull request by Tal-or**
In the current behavior, we're changing the ```LogLevel``` field under the CR and then checking the change results under the container.
There might be times when the reconciliation loop hasn't occurred yet (thus the pod didn't get updated) and we're already checking for the pod changes under the test.

With the new behavior using Eventually we'll have until timeout to check if the test condition were met.

Signed-off-by: Talor Itzhak <titzhak@redhat.com>

### CODE REVIEW

**Review by fromanirh — COMMENTED** [review]
using Eventually is more correct anyway, so I'm all for this change. Please make CI happy and ping me for the final review (spoiler: will likely LGTM immediately unless there are some major issues)

File: test/e2e/rte/rte_test.go [hunk 1]
```diff
@@ -73,21 +74,37 @@ var _ = ginkgo.Describe("with a running cluster with all the components", func()
...
+
+					// TODO better match by name than assumes container #0 is the right one
+					rteCnt := &ds.Spec.Template.Spec.Containers[0]
+					found, match := matchLogLevelToKlog(rteCnt, nropObj.Spec.LogLevel)
+					if !found {
+						klog.Warningf("DaemonSet: %s --v flag doesn't exist in container %s args under ds", ds.Name, rteCnt.Name)
+					}
+					isMatchedInAllDss = isMatchedInAllDss && (found && match)
+				}
+				return isMatchedInAllDss
```
Comment (fromanirh) [inline test/e2e/rte/rte_test.go:105]: would this work?
```
for _, ds := range rteDss {
	// TODO better match by name than assumes container #0 is the right one
	rteCnt := &ds.Spec.Template.Spec.Containers[0]
	found, match := matchLogLevelToKlog(rteCnt, nropObj.Spec.LogLevel)
	if !found {
		klog.Warningf("DaemonSet: %s --v flag doesn't exist in container %s args under ds", ds.Name, rteCnt.Name)
		return false
	}
	if !match {
        	return false
	}
}
return true
```

File: test/e2e/rte/rte_test.go [hunk 2]
```diff
@@ -98,17 +115,32 @@ var _ = ginkgo.Describe("with a running cluster with all the components", func()
...
+					rteCnt := &ds.Spec.Template.Spec.Containers[0]
+					found, match := matchLogLevelToKlog(rteCnt, nropObj.Spec.LogLevel)
+					if !found {
+						klog.Warningf("DaemonSet: %s --v flag doesn't exist in container %s args under ds", ds.Name, rteCnt.Name)
+						return false
+					}
+
+					isMatchedInAllDss = isMatchedInAllDss && (found && match)
+				}
+				return isMatchedInAllDss
```
Comment (fromanirh) [inline test/e2e/rte/rte_test.go:141]: same as above

**Review by fromanirh — COMMENTED** [review]
looks good, maybe we can simplify a bit more

File: test/e2e/rte/rte_test.go (hunk 1 above)
Comment (Tal-or) [reply test/e2e/rte/rte_test.go:105]: nice!

**Comment by Tal-or** [comment]
/retest
infra issue

**Review by fromanirh — COMMENTED** [review]
/approve
/lgtm

**Comment by Tal-or** [comment]
I flipped the positions of the interval and timeout arguments under the ```Eventually``` function. it should be ok now

**Review by fromanirh — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**